sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs

```diff
@@ -676,111 +676,141 @@ public async Task CanQueryWithFormattedQuery(FormattableStringWrapper query)
-                    verifyColumn2 = true;
-            }
+        [Test]
+        public async Task CanQueryResourcePrimaryWorkspace()
+        {
+            var timespan = TimeSpan.FromSeconds(5);
+
+            var client = CreateClient();
+            // Empty check
+            Response<LogsQueryResult> results = await client.QueryResourceAsync(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
+
+            Assert.AreEqual(0, results.Value.Table.Rows.Count);
+
+            // Check if all rows in table were uploaded
+            // Get the time of the third event and add a bit of buffer to it (events are 2d apart)
+            var maxOffset = (DateTimeOffset)_logsTestData.TableA[2][LogsTestData.TimeGeneratedColumnNameSent];
+            timespan = Recording.UtcNow - maxOffset;
+            timespan = timespan.Add(TimeSpan.FromDays(7));
+
+            // Make sure there is some data in the range specified
+            results = await client.QueryResourceAsync(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
+
+            Assert.GreaterOrEqual(results.Value.Table.Rows.Count, 3);
+        }
 
-            Assert.IsTrue(verifyRow);
-            Assert.IsTrue(verifyColumn1 && verifyColumn2);
+        [Test]
+        public async Task CanQueryResourceSecondaryWorkspace()
+        {
+            var timespan = TimeSpan.FromSeconds(5);
+
+            var client = CreateClient();
+            // Empty check
+            Response<LogsQueryResult> results = await client.QueryResourceAsync(
+                new ResourceIdentifier(TestEnvironment.WorkspaceSecondaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
+
+            Assert.AreEqual(0, results.Value.Table.Rows.Count);
+
+            // Check if all rows in table were uploaded
+            // Get the time of the third event and add a bit of buffer to it (events are 2d apart)
+            var maxOffset = (DateTimeOffset)_logsTestData.TableA[2][LogsTestData.TimeGeneratedColumnNameSent];
+            timespan = Recording.UtcNow - maxOffset;
+            timespan = timespan.Add(TimeSpan.FromDays(7));
+
+            // Make sure there is some data in the range specified
+            results = await client.QueryResourceAsync(
+                new ResourceIdentifier(TestEnvironment.WorkspaceSecondaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
+
+            Assert.GreaterOrEqual(results.Value.Table.Rows.Count, 3);
         }
 
-        [LiveOnly]
```

Comment:
Since these are no longer LiveOnly we will need to also include the session records for these tests.